src/Razor/test/RazorLanguage.Test/TestFiles/IntegrationTests/ComponentRuntimeCodeGenerationTest/BindToElementFallback_WritesAttributes/TestComponent.codegen.cs

```diff
@@ -24,7 +24,7 @@ protected override void BuildRenderTree(Microsoft.AspNetCore.Components.RenderTr
 #line hidden
 #nullable disable
             ));
-            builder.AddAttribute(3, "onchange", Microsoft.AspNetCore.Components.EventCallback.Factory.CreateBinder(this, __value => ParentValue = __value, ParentValue));
+            builder.AddAttribute(3, "onchange", Microsoft.AspNetCore.Components.EventCallback.Factory.CreateBinder(this, __value => ParentValue = __value, ParentValue), "value");
```

Comment:
Hmm curious, why would this information not be part of the event callback creation?

Reply:
It's super complicated. I will happily explain this in full next week.

Short summary is that at runtime we need to know whether a given event handler represents part of a two-way binding or not, and if so, which other attribute is the "value" half of the two-way binding.

The client-side code doesn't have this info in any robust way, and even if it were to guess, the server-side code doesn't want to trust the client to be truthful about it.